src/UI/Dependency/EnvironmentModifier.test.tsx

```diff
@@ -79,3 +85,21 @@ test("Given the environmentModifier When the server compile setting is requested
     await screen.findByRole("generic", { name: "server-compile-enabled" })
   ).toBeVisible();
 });
+
+test("Given the environmentModifier When the missing setting is requested Then returns false without crashing the component", async () => {
+  delete EnvironmentSettings.definition.server_compile;
+  const { component, store, environmentId } = setup(
+    EnvironmentSettings.definition
+  );
+  // No setting is specified, and the dafault is missing
+  store.dispatch.environment.setEnvironmentDetailsById({
+    id: environmentId,
+    value: RemoteData.success({ ...EnvironmentDetails.a, settings: {} }),
+  });
+
+  render(component);
+  //expect to see div as the value would be false insteand of error boundary page
```

Comment:
instead*
and see a div is really abstract, is it an empty div?

The test description also doesn't match what you are testing. You're not asserting a return value being false, but testing if a server-compile-disabled element is visible in the DOM. 

Also, being completely accurate, if you write in the comment that one should be displayed instead of another, you should also test that the other element is not present. Theoretically, both could be present in the DOM and this test could result in a fake passing test.

Reply:
the setup is made in a way that setting value is represented by the div, either with the name "disabled" or "enabled" in it and It couldn't be possible to have both in them but I'll add assesrtion as it doesn't cost a thing. I'll add also assertion for consol.error to have errrorBoundary component covered

Reply:
Yes, but setup can change :) 
About the div, it's more the comment not being descriptive enough, a div could be anything :)